### ISSUE TITLE
TTRPC Error

### ISSUE THREAD

**Issue by g0ldingkev** [VMwareFusion/nautilus#4]
any specific requirements on the version of fusion running on the mac, or any docs?.  tried creating a container and running it, at runtime i get the following:

ERROR failed when running container my-www: ttrpc: closed: unknown

**Comment by mkangvmware** [comment]
what's your macOS version? could you please share the container you are trying to run?

**Comment by jeffwubj** [comment]
this is a getting start doc https://github.com/VMwareFusion/nautilus/blob/master/docs/getting-started.md
would you mind sharing what image you are running ? also the output of:
```
vctl list c -a
```

there will be a log file under ~/.nautilus/containerd.log, by default, it is of info level, following commands
```
vctl system stop
vctl system start -l debug

```
will write debug level logs in ~/.nautilus/containerd.log, share that will also be helpful.

**Comment by g0ldingkev** [comment]
thanks guys.  so MAC OS version is Mojave 10.14.5.  The container trying to run is the one in the blog post, docker.io/mikeroysoft/mrs-hugo:dev.

Container is here...

Kevins-MacBook-Air:~ kevingolding$ vctl list c -a     
NAME     IMAGE                                COMMAND                IP              PORTS   STATUS    CREATION TIME          
      
my-www   docker.io/mikeroysoft/mrs-hugo:dev   nginx -g daemon off;   not available           stopped   2020-01-22T08:15:06Z 

I did the suggestions above, it seems to be a permissions issue, but the vctl start command asks for admin permissions.

i've outputted the containerd.log file into the attached doc.

[Untitled.pdf](https://github.com/VMwareFusion/nautilus/files/4097154/Untitled.pdf)

**Comment by yeahdongcn** [comment]
I have seen this before and it was caused by my Fusion 11.5.1 was running. After quitting Fusion 11.5.1, everything back to normal.

**Comment by g0ldingkev** [comment]
thanks @yeahdongcn, i've resolved it..  Fusion was definitely stopped.  but it seems that nautilus wanted to add an exception to the security settings.  the initial couple of times of trying the dialog box didnt appear (or was hidden)...  after i cleared everything down the dialog box appeared when trying to start the container.

all working now.  thanks for the help.

**Comment by gerhard** [comment]
TL;DR Restarting Fusion Tech Preview (I don't have any other Fusion installed) **and then restarting containerd** via `vctl system stop && vctl system start` fixed this issue for me.

I'm running:

```
neofetch

                    'c.          gerhard@focker
                 ,xNMM.          --------------
               .OMMMMo           OS: macOS Mojave 10.14.6 18G4032 x86_64
               OMMM0,            Host: iMacPro1,1
     .;loddo:' loolloddol;.      Kernel: 18.7.0
   cKMMMMMMMMMMNWMMMMMMMMMM0:    Uptime: 1 day, 18 hours, 31 mins
 .KMMMMMMMMMMMMMMMMMMMMMMMWd.    Packages: 204 (brew)
 XMMMMMMMMMMMMMMMMMMMMMMMX.      Shell: bash 3.2.57
;MMMMMMMMMMMMMMMMMMMMMMMM:       Resolution: 2880x1620@2x
:MMMMMMMMMMMMMMMMMMMMMMMM:       DE: Aqua
.MMMMMMMMMMMMMMMMMMMMMMMMX.      WM: Quartz Compositor
 kMMMMMMMMMMMMMMMMMMMMMMMMWd.    WM Theme: Blue (Dark)
 .XMMMMMMMMMMMMMMMMMMMMMMMMMMk   Terminal: iTerm2
  .XMMMMMMMMMMMMMMMMMMMMMMMMK.   Terminal Font: MesloLGSNerdFontComplete-Regular 12
    kMMMMMMMMMMMMMMMMMMMMMMd     CPU: Intel Xeon W-2150B (20) @ 3.00GHz
     ;KMMMMMMMWXXWMMMMMMMk.      GPU: Radeon Pro Vega 56
       .cooc,.    .,coo:.        Memory: 16495MiB / 65536MiB

vctl version: e.x.p
containerd github.com/containerd/containerd v1.3.2
```

![image](https://user-images.githubusercontent.com/3342/77927069-cabcee00-729e-11ea-88f3-f5b63ea71f9a.png)

The initial error message didn't mean anything to me:

```
vctl run c rmq0 -i docker.io/library/rabbitmq:3.8.3-management
ERROR failed when running container rmq0: ttrpc: closed: unknown
```

After I stopped Fusion (containerd was still running), I got the same error, but formatted differently:

```
vctl run container rmq0 -i docker.io/library/rabbitmq:3.8.3-management
ERROR resize pty                                    error="ttrpc: closed: unknown"
```

Having restarted Fusion, vctl was incorrectly reporting the container as started:

```
vctl run c rmq1 -i docker.io/library/rabbitmq:3.8.3-management
INFO container rmq1 started
vctl ls c
────   ─────   ───────   ──   ─────   ──────   ─────────────
NAME   IMAGE   COMMAND   IP   PORTS   STATUS   CREATION TIME
────   ─────   ───────   ──   ─────   ──────   ─────────────
$ vctl ls c -a
────   ─────                                         ───────                   ──              ─────   ──────    ─────────────
NAME   IMAGE                                         COMMAND                   IP              PORTS   STATUS    CREATION TIME
────   ─────                                         ───────                   ──              ─────   ──────    ─────────────
rmq0   docker.io/library/rabbitmq:3.8.3-management   docker-entrypoint.sh...   not available           stopped   2020-03-30T15:48:08+01:00
rmq1   docker.io/library/rabbitmq:3.8.3-management   docker-entrypoint.sh...   not available           stopped   2020-03-30T15:48:47+01:00
```

Starting the container explicitly didn't work either:

```
vctl start c rmq1
INFO container rmq1 started
$ vctl ls c -a
────   ─────                                         ───────                   ──              ─────   ──────    ─────────────
NAME   IMAGE                                         COMMAND                   IP              PORTS   STATUS    CREATION TIME
────   ─────                                         ───────                   ──              ─────   ──────    ─────────────
rmq0   docker.io/library/rabbitmq:3.8.3-management   docker-entrypoint.sh...   not available           stopped   2020-03-30T15:48:08+01:00
rmq1   docker.io/library/rabbitmq:3.8.3-management   docker-entrypoint.sh...   not available           stopped   2020-03-30T15:48:47+01:00
```

Only after I stopped & started containerd, it all started working correctly:

```
vctl start container rmq1
INFO container rmq1 started
2020-03-30 15:51:45.590 [info] <0.9.0> Feature flags: list of feature flags found:
...
2020-03-30 15:53:00.984 [info] <0.9.0> Server startup complete; 3 plugins started.
 * rabbitmq_management
 * rabbitmq_web_dispatch
 * rabbitmq_management_agent
```

Thanks @cdickmann for mentioning Nautilius earlier. I am so looking forward to finally replacing Docker for Desktop.

@michaelklishin is a fan of Fusion and will be so pleased to learn that he can now run **any** RabbitMQ Docker image on [Fusion Tech Preview](https://blogs.vmware.com/teamfusion/2020/01/fusion-tp20h1-introducing-nautilus.html) in 2 commands, e.g.

```
vctl pull image docker.io/library/rabbitmq:3.8.3-management
vctl run container rmq0 -i --hostname rmq0 --image docker.io/library/rabbitmq:3.8.3-management
```

**Comment by mikeroySoft** [comment]
@jeffwubj Is this issue still valid with vctl 1.0?
